test/e2e/network/networking_perf.go

```diff
@@ -160,3 +171,198 @@ var _ = SIGDescribe("Networking IPerf [Experimental] [Slow] [Feature:Networking-
+	if err != nil {
+		return nil, fmt.Errorf("deployment %q Create API error: %v", deploymentSpec.Name, err)
+	}
+	framework.Logf("Waiting deployment %q to complete", deploymentSpec.Name)
+	err = e2edeployment.WaitForDeploymentComplete(client, deployment)
+	if err != nil {
+		return nil, fmt.Errorf("deployment %q failed to complete: %v", deploymentSpec.Name, err)
+	}
+	return deployment, nil
+}
+
+func iperf3ServerService(client clientset.Interface, namespace string) (*v1.Service, error) {
+	service := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{Name: "iperf3-server"},
+		Spec: v1.ServiceSpec{
+			Selector: map[string]string{
+				"app": "iperf3-server",
+			},
+			Ports: []v1.ServicePort{
+				{Protocol: v1.ProtocolTCP, Port: 5201},
+			},
+		},
+	}
+	return client.CoreV1().Services(namespace).Create(context.TODO(), service, metav1.CreateOptions{})
+}
+
+func iperf3ClientDaemonSet(client clientset.Interface, namespace string) (*appsv1.DaemonSet, error) {
+	zero := int64(1)
+	labels := map[string]string{"app": "iperf3-client"}
+	spec := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "iperf3-clients",
+			Labels: labels,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: v1.PodSpec{
+					Tolerations: []v1.Toleration{
+						{
+							Key:      "node-role.kubernetes.io/master",
+							Operator: v1.TolerationOpEqual,
+							Effect:   v1.TaintEffectNoSchedule,
+						},
+					},
+					Containers: []v1.Container{
+						{
+							Name:    "iperf3-client",
+							Image:   imageutils.GetE2EImage(imageutils.Iperf3),
+							Command: []string{"/bin/sh"},
+							Args:    []string{"-c", "sleep infinity"},
+						},
+					},
+					TerminationGracePeriodSeconds: &zero,
+				},
+			},
+		},
+		Status: appsv1.DaemonSetStatus{},
+	}
+
+	ds, err := client.AppsV1().DaemonSets(namespace).Create(context.TODO(), spec, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("daemonset %q Create API error: %v", spec.Name, err)
+	}
+	return ds, nil
+}
+
+type iperf3Results struct {
+	End struct {
+		SumSent struct {
+			Start         float64 `json:"start"`
+			End           float64 `json:"end"`
+			Seconds       float64 `json:"seconds"`
+			Bytes         float64 `json:"bytes"`
+			BitsPerSecond float64 `json:"bits_per_second"`
+			Retransmits   int     `json:"retransmits"`
+		} `json:"sum_sent"`
+		SumReceived struct {
+			Start         float64 `json:"start"`
+			End           float64 `json:"end"`
+			Seconds       float64 `json:"seconds"`
+			Bytes         float64 `json:"bytes"`
+			BitsPerSecond float64 `json:"bits_per_second"`
+		} `json:"sum_received"`
+	} `json:"end"`
+}
+
+var _ = SIGDescribe("Networking IPerf3 IPv4 [Experimental] [Feature:Networking-IPv4] [Slow] [Feature:Networking-Performance]", func() {
```

Comment:
this is already running in the CI for ipv4 and for ipv6
https://testgrid.k8s.io/sig-network-kind#sig-network-kind,%20IPv6,%20master , it must work in both

Reply:
Ah good call, I'll make the same change as in the existing test to support both!

Reply:
👍